### PR TITLE
pidgin-indicator: init at 1.0.1

### PR DESCRIPTION
**Pull request by colonelpanic8**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/pidgin-indicator/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-CdA/aUu+CmCRbVBKpJGydicqFQa/rEsLWS3MBKlH2/M=";
+  };
+
+  nativeBuildInputs = [ autoreconfHook ];
+  buildInputs = [ glib intltool libappindicator-gtk2 libtool pidgin ];
+
+  meta = {
+    description = "A AppIndicator/KStatusNotifierItem Plugin for Pidgin";
```
Comment (legendofmiracles) [inline pkgs/applications/networking/instant-messengers/pidgin-plugins/pidgin-indicator/default.nix:27]:
```suggestion
    description = "An AppIndicator/KStatusNotifierItem Plugin for Pidgin";
```

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/pidgin-indicator/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    owner = "philipl";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-CdA/aUu+CmCRbVBKpJGydicqFQa/rEsLWS3MBKlH2/M=";
+  };
+
+  nativeBuildInputs = [ autoreconfHook ];
+  buildInputs = [ glib intltool libappindicator-gtk2 libtool pidgin ];
+
+  meta = {
```
Comment (legendofmiracles) [inline pkgs/applications/networking/instant-messengers/pidgin-plugins/pidgin-indicator/default.nix:26]:
```suggestion
  meta =  with lib; {
```

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/pidgin-indicator/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    description = "A AppIndicator/KStatusNotifierItem Plugin for Pidgin";
+    homepage = "https://github.com/philipl/pidgin-indicator";
+    maintainers = with lib.maintainers; [ imalison ];
```
Comment (legendofmiracles) [inline pkgs/applications/networking/instant-messengers/pidgin-plugins/pidgin-indicator/default.nix:29]:
```suggestion
    maintainers = with maintainers; [ imalison ];
```

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/pidgin-indicator/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    license = lib.licenses.gpl2;
```
Comment (legendofmiracles) [inline pkgs/applications/networking/instant-messengers/pidgin-plugins/pidgin-indicator/default.nix:30]:
```suggestion
    license = licenses.gpl2;
```
gpl2Plus or gpl2Only?

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/pidgin-indicator/default.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    platforms = with lib.platforms; linux;
```
Comment (legendofmiracles) [inline pkgs/applications/networking/instant-messengers/pidgin-plugins/pidgin-indicator/default.nix:31]:
```suggestion
    platforms = platforms.linux;
```
Should also run on other platforms as well, no?

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/pidgin-indicator/default.nix (hunk 1 above)
Comment (IvanMalison) [reply pkgs/applications/networking/instant-messengers/pidgin-plugins/pidgin-indicator/default.nix:27]: I mean its not clear which should be used. An does not work with KStatusNotifierItem. Seems like either choice is valid to me.

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/pidgin-indicator/default.nix (hunk 5 above)
Comment (IvanMalison) [reply pkgs/applications/networking/instant-messengers/pidgin-plugins/pidgin-indicator/default.nix:31]: its specifically for a StatusNotifierItem indicator. I suppose that maybe technically that protocol is supported on non-linux platforms (free bsd hurd maybe), but I think I've seen other packages mostly specify linux for dbus/appindicator based stuff.

Certainly not usuable in e.g. darwin.

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/pidgin-indicator/default.nix (hunk 4 above)
Comment (IvanMalison) [reply pkgs/applications/networking/instant-messengers/pidgin-plugins/pidgin-indicator/default.nix:30]: I'm not sure how to tell. I just saw gpl2 here:

https://github.com/philipl/pidgin-indicator/blob/master/COPYING

**Pull request merged**